### PR TITLE
docs: add nocloud documentation

### LINKED ISSUE #4354: Document nocloud support
#4208 added support for nocloud as a platform, but there is no documentation on how to use it.

From the original PR:
> @sergelogvinov doesn't have to be in this PR, but it'd be awesome if you could add a bit to the docs about what NoCloud is, when you would use the nocloud image, and maybe how you'd do it w/ proxmox or something.

https://github.com/talos-systems/talos/pull/4208#pullrequestreview-765495122

### PR DESCRIPTION
**Pull request by sergelogvinov**
Update documentation - how to use nocloud platform.

Close #4354

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4377)

### CODE REVIEW

File: website/content/docs/v0.13/Cloud Platforms/nocloud.md [hunk 1]
```diff
@@ -3,4 +3,110 @@ title: "Nocloud"
...
+
+* SMBIOS “serial number” option
+* CDROM or USB-flash filesystem
+
+### SMBIOS
+
+This method works only with network.
+DHCP and HTTP(s) server required.
+
+```
```
[the remark this answers is not in the document]
Comment (sergelogvinov) [reply website/content/docs/v0.13/Cloud Platforms/nocloud.md:17]: Yep, I've add the screenshot bellow

File: website/content/docs/v0.13/Cloud Platforms/nocloud.md [hunk 2]
```diff
@@ -3,4 +3,121 @@ title: "Nocloud"
...
+-smbios type=1,serial=ds=nocloud-net;s=http://10.10.0.1/configs/
+```
+
+#### Proxmox
+
+Set the source machine config throught the serial number on Proxmox GUI.
+
+<img src="/images/proxmox-guide/proxmox-smbios.png" width="920px">
```
Comment (kurious-diru) [inline website/content/docs/v0.13/Cloud Platforms/nocloud.md:38]: wonder if it is easier to identify if it is in a new no-cloud directory

File: website/content/docs/v0.13/Cloud Platforms/nocloud.md [hunk 3]
```diff
@@ -3,4 +3,121 @@ title: "Nocloud"
...
+
+The Proxmox stores the VM config here /etc/pve/qemu-server/$ID.conf (```$ID``` - VM ID number of virtual machine), you will see someting like:
```
Comment (kurious-diru) [inline website/content/docs/v0.13/Cloud Platforms/nocloud.md:40]:
```suggestion
The Proxmox stores the VM config at /etc/pve/qemu-server/$ID.conf (```$ID``` - VM ID number of virtual machine), you will see something like:
```

File: website/content/docs/v0.13/Cloud Platforms/nocloud.md [hunk 4]
```diff
@@ -3,4 +3,121 @@ title: "Nocloud"
...
+...
+smbios1: uuid=ceae4d10,serial=ZHM9bm9jbG91ZC1uZXQ7cz1odHRwOi8vMTAuMTAuMC4xL2NvbmZpZ3Mv,base64=1
+...
+```
+
+Where serial is a base64-encoded string of `ds=nocloud-net;s=http://10.10.0.1/configs/`.
+
+### CDROM/USB
+
+Talos can also get machine config from local storage without running a network services (DHCP/HTTP).
```
Comment (kurious-diru) [inline website/content/docs/v0.13/Cloud Platforms/nocloud.md:52]:
```suggestion
Talos can also get machine config from local storage without running network services (DHCP/HTTP).
```

File: website/content/docs/v0.13/Cloud Platforms/nocloud.md [hunk 5]
```diff
@@ -3,4 +3,121 @@ title: "Nocloud"
...
+    ...
+    -cdrom iso/cidata.iso \
+    ...
+```
+
+#### Proxmox
+
+Proxmox can create cloud-init disk for you.
+[Original documentation](https://pve.proxmox.com/wiki/Cloud-Init_Support).
+Edit the config file as follows, substituting your own infomation as necessary:
```
Comment (kurious-diru) [inline website/content/docs/v0.13/Cloud Platforms/nocloud.md:105]:
```suggestion
Edit the cloud-init config information in proxmox as follows,  substitute your own information as necessary:
```

**Review by kurious-diru — CHANGES_REQUESTED** [review]

**Review by kurious-diru — APPROVED** [review]

**Comment by Born2Bake** [comment]
I noticed that when you do something within network.interfaces, talos overwrites settings set in ipconfig0 therefore, it erases networking settings in cloud-init. Thus, if you need to set vip ip address, talos erases settings set in cloud-init ipconfig0 

:disappointed:

```
  network:
    hostname: "${hostname}"
    interfaces:
      - interface: eth0
        addresses:
          - ${ipv4_local}/24
        routes:
          - network: 0.0.0.0/0
            gateway: ${gateway}
        vip:
          ip: ${ipv4_vip}
```

**Comment by smira** [comment]
> I noticed that when you do something within network.interfaces, talos overwrites settings set in ipconfig0 therefore, it erases networking settings in cloud-init. Thus, if you need to set vip ip address, talos erases settings set in cloud-init ipconfig0

#4755 should actually address that, Talos will stop patching machine configuration

**Review by smira — APPROVED** [review]

**Pull request merged**